### PR TITLE
Upgrade to Ruby 3.1.2

### PR DESCRIPTION
**Pull request by raccube**

### CODE REVIEW

**Review by pixeldesu — APPROVED** [review]

File: lib/use_case/base.rb [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
-    def self.call(*args, **kwargs)
-      new(*args, **kwargs).call
+    def self.call(...)
+      new(...).call
     end
 
     def call
       raise NotImplementedError
     end
```
Comment (nilsding) [inline lib/use_case/base.rb]: You can even do one of my favourite things in Ruby 3:

```suggestion
    def self.call(...) = new(...).call

    def call = raise NotImplementedError
```

**Review by nilsding — APPROVED** [review]

File: lib/use_case/base.rb [hunk 2]
```diff
@@ -1,19 +1,15 @@
...
 
-    def self.call(*args, **kwargs)
-      new(*args, **kwargs).call
-    end
+    def self.call(...) = new(...).call
 
-    def call
-      raise NotImplementedError
-    end
+    def call = raise NotImplementedError
```
[the remark this answers is not in the document]
Comment (raccube) [reply lib/use_case/base.rb:13]: We may have to update something here…

[the remark this answers is not in the document]
Comment (nilsding) [reply lib/use_case/base.rb:13]: yeah, the `rubocop.yml`

**Pull request merged**